python/lsst/meas/astrom/matchPessimisticB.py

```diff
@@ -472,23 +476,82 @@ def _doMatch(self, refCat, sourceCat, wcs, refFluxField, numUsableSources,
                 match_tolerance.lastMatchedPattern = matcher_struct.pattern_idx
                 break
 
+        # The matcher returns all the nearst neighbors between the reference
+        # source catalogs. For the currrent astrometry solver we need to clip
+        # these down to give the best solution.
+        distances = np.degrees(matcher_struct.distances) * 3600
+        dist_cut = np.max(
+            (2 * wcs.pixelScale().asArcseconds(),
+             self._sigma_clip(
+                distances, 2,
+                np.max((10 * wcs.pixelScale().asArcseconds(),
+                        maxMatchDistArcSec,
+                        maxShiftArcseconds)))))
+
         # A match has been found, return our list of matches and
         # return.
         matches = []
         for match_ids, dist in zip(matcher_struct.matches,
-                                   matcher_struct.distances):
-            match = afwTable.ReferenceMatch()
-            match.first = refCat[match_ids[1]]
-            match.second = sourceCat[match_ids[0]]
-            match.distance = match.first.getCoord().angularSeparation(
-                match.second.getCoord())
-            matches.append(match)
+                                   distances):
+            if dist < dist_cut:
+                match = afwTable.ReferenceMatch()
+                match.first = refCat[match_ids[1]]
+                match.second = sourceCat[match_ids[0]]
+                match.distance = match.first.getCoord().angularSeparation(
+                    match.second.getCoord())
+                matches.append(match)
 
         return pipeBase.Struct(
             matches=matches,
             match_tolerance=match_tolerance,
         )
 
+    def _sigma_clip(self, dists, n_sigma, input_cut=None):
+        """
+        Find the distance cut by recursively cliping by n_sigma.
+
+        The algorithm converges once the length of the array on the
+        nth iteration is equal to the the length of the array on the
+        n+1th interation.
+
+        Parameters
+        ----------
+        dists : float array
+            float array distances to caculate the cut on.
```

Comment:
Please consider `values`; it could be distances, but it could be anything. If you use distances then the next obvious question users will ask is "what units" (that was my first reaction until I realized it didn't matter).